In [15]:
import pandas as pd
import numpy as np

#### Dataset: Amazon Prime TV shows and titles.

- The goal of this project is exploratory data analysis to potentially derive interesting and valuable insights about Amazon TV shows and titles.
- Two (2) csv files are available for use in this project and  are derived from the Kaggle (link is provided below)
- Source of data: https://www.kaggle.com/datasets/victorsoeiro/amazon-prime-tv-shows-and-movies?resource=download&select=titles.csv

In [16]:
# Load the csv files Amazon Prime TV shows
tv_credits = pd.read_csv(r'C:\Users\LuisV\Desktop\UDACITY\Portfolio\Amazon Prime TV shows and titles\amazon_project\credits.csv')

In [17]:
# Display the first 5 rows of the dataframe
tv_credits.head(5)

person_id       id           name      character   role
0      59401  ts20945     Joe Besser            Joe  ACTOR
1      31460  ts20945     Moe Howard            Moe  ACTOR
2      31461  ts20945     Larry Fine          Larry  ACTOR
3      21174  tm19248  Buster Keaton    Johnny Gray  ACTOR
4      28713  tm19248    Marion Mack  Annabelle Lee  ACTOR

In [18]:
# Display the number of rows and columns in the dataframe
tv_credits.shape

(124235, 5)

In [48]:
# Create a copy of the dataframe
tv = tv_credits.copy()

In [49]:
# Check for the number of rows and columns in the copy of the dataframe
tv.shape

(124235, 5)

In [50]:
# Check for the total number of duplicate rows in the data
tv.duplicated().sum()

56

- There are 56 rows duplicate rows in the dataframe

In [51]:
# Drop all duplicate rows across all columns
tv = tv.drop_duplicates()

In [52]:
# Check for the number of rows and columns after removing the duplicates
tv.shape

(124179, 5)

- There are 124,179 rows left after removal of duplicate rows across all columns

In [53]:
# Check for the total number of duplicate rows in the dataframe
tv.duplicated().sum()

0

In [54]:
# Check for the number of null values in the dataframe
tv.isnull().sum()

person_id        0
id               0
name             0
character    16277
role             0
dtype: int64

- There are 16,277 empty rows in the Character column, while other columns have no empty entry.

In [55]:
# Fill missing values in the Character column with a forward fill method
tv = tv.fillna(method = 'ffill', axis = 0)

In [56]:
# Check for the number of null values in the dataframe
tv.isnull().sum()

person_id    0
id           0
name         0
character    0
role         0
dtype: int64

- There are no null values left in the dataframe